Assets/Scripts/Control&Input/PlayerInputManagerController.cs

```diff
@@ -48,10 +48,10 @@ public void RemoveListeners()
 
     private void OnPlayerJoined(PlayerInput playerInput)
     {
+        playerInputs.Add(playerInput);
         var playerIdentity = playerInput.GetComponent<PlayerIdentity>();
-        playerIdentity.color = playerColors[playerInputs.Count];
+        playerIdentity.color = playerColors[playerInputs.Count-1];
         playerIdentity.playerName = playerNames[playerInputs.Count];
```

Comment:
@Fueredoriku doesn't the indexing issue affect this as well?

Reply:
One would think so, but my testing proved otherwise

Reply:
Haha I was right. Fixed it along with some other issues.